src/main/java/com/capitalone/dashboard/event/TestResultEventListener.java

```diff
@@ -0,0 +1,281 @@
+    }
+
+    @Autowired
+    public TestResultEventListener(CollectorRepository collectorRepository, CollectorItemRepository collectorItemRepository,
+                                   PerformanceRepository performanceRepository) {
+        this.collectorRepository = collectorRepository;
+        this.collectorItemRepository = collectorItemRepository;
+        this.performanceRepository = performanceRepository;
+    }
+
+    /**
+     * Create and save performance object for every test_result type is performance
+     * @param event
+     */
+    @Override
+    public void onAfterSave(AfterSaveEvent<TestResult> event) {
+        TestResult testResult = event.getSource();
+        LOGGER.info("TestResult save event triggered");
+
+        // Ignore anything other than performance tests
+        if (!TestSuiteType.Performance.equals(testResult.getType())) {
+            LOGGER.info("TestResult save event ignored since its not performance type");
+            return;
+        }
+
+        List<TestCapability> testCapabilities = new ArrayList<>(testResult.getTestCapabilities());
+        if (CollectionUtils.isEmpty(testCapabilities)){
+            return;
+        }
+        testCapabilities.sort(Comparator.comparing(TestCapability::getTimestamp).reversed());
+        TestCapability lastExecutedTestCapabiblity = testCapabilities.iterator().next();
+        lastExecutedTestCapabiblity.getTestSuites().forEach(testSuite -> testSuite.getTestCases().forEach(testCase ->
+                readPerformanceMetrics(testCase, lastExecutedTestCapabiblity)));
+        CollectorItem perfCollectorItem = getPerfCollectorItem(testResult);
+        createPerformanceDoc(testResult, lastExecutedTestCapabiblity, perfCollectorItem);
+        LOGGER.info("New performance document created from test_result successfully");
+        }
+
+    /**
+     * Creates new performance object from test result
+     * @params testResult, testCapability, perfCollectorItem
+     */
+    public Performance createPerformanceDoc(TestResult testResult, TestCapability testCapability, CollectorItem perfCollectorItem) {
+        Performance performance = new Performance();
+        performance.setId(ObjectId.get());
+        performance.setCollectorItemId(perfCollectorItem.getId());
+        performance.setType(PerformanceType.ApplicationPerformance);
+        performance.setUrl(testResult.getUrl());
+        performance.setTimestamp(System.currentTimeMillis());
+        performance.setTargetAppName(testResult.getTargetAppName());
+        performance.setTargetEnvName(testResult.getTargetEnvName());
+        performance.setMetrics(getPerfMetrics(testCapability));
+        return performanceRepository.save(performance);
+    }
+    /**
+     * Creates new performance collector item
+     * @params testResult
+     */
+    public CollectorItem getPerfCollectorItem(TestResult testResult) {
+
+        CollectorItem testResultCItem = collectorItemRepository.findOne(testResult.getCollectorItemId());
```

Comment:
rename all variables that have *CItem to *ColItem
example testResultCItem to testResultCollItem
we use CI inside Hygieia cmdb so this helps avoids any misunderstanding

Reply:
renamed as mentioned